crates/turborepo-lib/src/shim.rs

```diff
@@ -750,7 +769,7 @@ pub fn run() -> Result<i32, Error> {
     // global turbo having handled the inference. We can run without any
     // concerns.
     if args.skip_infer {
-        return Ok(cli::run(None, &subscriber, ui)?);
+        return Ok(cli::run(None, &subscriber, ui, cli::BinaryType::SkipInfer)?);
```

Comment:
I think this is where we want to check the env var and decide between `SkipInfer` and `LocalDefer`